interface/src/app/node/node.js

```diff
@@ -304,6 +307,32 @@ angular.module('adage.node', [
         $scope.numExpShown =
           $scope.topMode ? $scope.topNum : $scope.experiments.length;
       };
+
+      // Event handler when user clicks the checkbox in front of each sample.
+      // It keeps track of the number of selected samples in each experiment
+      // to indicate whether the "show annotations" button should be enabled.
+      $scope.toggleSelection = function(exp, selected) {
+        if (selected) {
+          ++exp.numSelections;
+        } else {
+          --exp.numSelections;
+        }
+      };
```

Comment:
Probably from a modern CPU's point of view, 6000 vs. 1 is not really a big deal. It will just crunch them before human being can sense the difference of its processing time.